libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -33,12 +33,30 @@ using namespace dev;
 using namespace langutil;
 using namespace dev::solidity;
 
+namespace {
+
+vector<ContractDefinition const*> resolveBaseContracts(ContractDefinition const& _contract)
```

Comment:
I wonder whether you really need this helper function - you only need this once anyways, don't you? And there you might as well walk over the inheritance specifiers directly?

Reply:
It's used twice in the `getbase{Modifier,Function}()` functions. And optionally in the `checkIllegalOverrides()` function if we decide to require the explicit redeclaration of override

Reply:
And i specifically need the contracts as i iterate over the functions

Reply:
You could just iterate over the InheritanceSpecifier's directly instead... and I'm still not entirely sure if the function and modifier parts couldn't be merged... but yeah - fine to keep it like it is.

Reply:
Even much simpler, if you use ``richIdentifier``, which is explicitly defined to compare equal if and only if the types compare equal.